### DISCUSSION TITLE
ENVIRONMENT_FALLBACK

### DISCUSSION THREAD

**Discussion by Emiliano-Bucci** [amannn/next-intl#670] [Q&A]
After upgrading to latest version (3.1.4) I've started to get this error 
`s [Error]: ENVIRONMENT_FALLBACK`
`code: 'ENVIRONMENT_FALLBACK',`
 during static build generation (all my pages exports getServerSideProps) so I guess it could be the 404 page (?)

The stack trace is -> `/node_modules/use-intl/dist/production/react.js:1:1332`

Any clues?

**Comment by TudorStegarescu** [comment]
The same issue with 3.2.4 (timeZone is defined in getRequestConfig)

**Reply by amannn** [reply]
This might be relevant for you from [the time zone configuration docs](https://next-intl-docs.vercel.app/docs/usage/configuration#time-zone):

> The time zone in Client Components is automatically inherited from the server side if you wrap the relevant components in a `NextIntlClientProvider` that is rendered by a Server Component. For all other cases, you can specify the value explicitly.

**Reply by TudorStegarescu** [reply]
based on documentation i cannot see requirements to wrap in NextIntlClientProvider, it was mentioned if we like, but not required.

**Reply by TudorStegarescu** [reply]
by adding timeZone prop to NextIntlClientProvider solved issues.
thanks @amannn

**Comment by brookslyrette** [comment]
I'd expect a `null` timezone to not error and just inherit the servers timezone.

**Reply by amannn** [reply]
From [the time zone docs](https://next-intl-docs.vercel.app/docs/usage/configuration#time-zone):
> The time zone in Client Components is automatically inherited from the server side if you wrap the relevant components in a `NextIntlClientProvider` that is rendered by a Server Component. For all other cases, you can specify the value explicitly.

Does that help?

**Reply by Ncookiez** [reply]
The docs make it seem like this is an option. If you don't specify a timezone you get that error. 

Even in the prop descriptions:

![image](https://github.com/amannn/next-intl/assets/22108522/68875268-cfcb-4956-98dc-050655dbe55b)

I am OK with using the user time zone in this scenario. The error should not be displayed.

**Comment by appleseed-iii** [comment]
doing this works

```ts
const timeZone = Intl.DateTimeFormat().resolvedOptions().timeZone;
return (
  <NextIntlClientProvider timeZone={timeZone}>
```
